src/Gui/MainWindow.h

```diff
@@ -201,7 +202,11 @@ class GuiExport MainWindow : public QMainWindow
     void updateActions(bool delay = false);
 
     enum StatusType {None, Err, Wrn, Pane, Msg, Log, Tmp, Critical};
-    void showStatus(int type, const QString & message);
+    void showStatus(int type, const QString& message);
+
+    void showHint(InputHint hint);

```

Comment:
```suggestion
    void showHint(const InputHint& hint);
```
What about const? Idk if it's efficient to pass as reference, or if this is less than the 4 word size, that @hyarion once mentioned?

Reply:
IDE suggested to me that it is - but it might be with older version, I'll check it out.

Reply:
Of course it is extremely difficult to do compile time printing:
```cpp
char (*__kaboom)[sizeof( YourTypeHere )] = 1;
void kaboom_print( void )
{
    printf( "%d", __kaboom );
}
```

Reply:
But if reference, we can forward decalre :')

Reply:
Agreed with `const` - I missed that - thanks.

As for the reference vs value here. It was strange for me too, I was taught that structs should always be passed by reference due to copying being slower and not needed in most cases. As it turns out, that is not the case. If the struct is small enough (basically that it can be stored in two registers) it is faster than passing it by const reference as then you have indirect access to memory which is slower than accessing registers.

@tritao checked it with disassembly and benchmark: https://github.com/FreeCAD/FreeCAD/pull/19636#issuecomment-2675828223 and it indeed seems to be the case at least for NOINLINE case (which kinda makes sense, as with inline copy still must happen and if it is passed by ref it may just reuse the same registers).

More info: https://www.reddit.com/r/C_Programming/comments/19dntzp/comment/kj87hgt/

Reply:
The struct in question does contain a more things than enum, namely the message and set of key sequences, that is well defined both in code and in the PR description. I agree with your point about silently growing in size - I marked it as pass by value when it was much simpler and consisted of `const char*` and enum reference and so it was fitting in one register. Now I'm no longer sure and it has potential to growth (contrary to color in mentioned PR) - I'll go with that suggestion.